test/embedded/generic-error.swift

```diff
@@ -0,0 +1,25 @@
+// RUN: %target-swift-emit-ir -parse-as-library -module-name main -verify %s -enable-experimental-feature Embedded -swift-version 5 -wmo -o /dev/null
+
+// REQUIRES: swift_in_compiler
+// REQUIRES: optimized_stdlib
+// REQUIRES: swift_feature_Embedded
+
+// We don't support calling a default witness method (which is generic) with dynamic self.
+
+public protocol P {
+  init(x: Int)
+}
+
+extension P {
+  init(y: Int) {
+    self.init(x: y)
+  }
+}
+
+public class C: P {
+  public required init(x: Int) {}
+
+  public convenience init(z: Int) {
+    self.init(y: z)  // expected-error {{cannot specialize generic function or default protocol method because it's called with a dynamic Self type}}
```

Comment:
There are no references to `Self` in this code. Could the error message be expanded to explain where exactly dynamic `Self` is involved and how the code could be rewritten so that it compiles?